### PR TITLE
Make base point values language specific

### PR DESCRIPTION
**Pull request by BlakeWilliams**
This extracts the base points to be a responsibility of each analyzer to
provide since each language can have different needs based on the
generated s-expressions.

This also puts us in a good place if we decide to make points part of
the configuration.

### CODE REVIEW

File: lib/cc/engine/analyzers/python/node.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ module Engine
     module Analyzers
       module Python
         class Node
-          SCRUB_PROPERTIES = ["_type", "attributes"].freeze
```
Comment (mxie) [inline lib/cc/engine/analyzers/python/node.rb]: Is this a related change?

Comment (BlakeWilliams) [inline lib/cc/engine/analyzers/python/node.rb]: Nope, this is me trying to trim down the resulting s-expressions since they have a heavy mass.

Comment (mxie) [inline lib/cc/engine/analyzers/python/node.rb]: Can we make a separate commit for this instead then?

Comment (BlakeWilliams) [inline lib/cc/engine/analyzers/python/node.rb]: Sure, I'll split it into it's own commit, but same PR.

**Comment by mxie** [comment]
How do we use `BASE_POINTS` currently? And should we do something similar for `violation.rb`?

File: lib/cc/engine/analyzers/python/main.rb [hunk 2]
```diff
@@ -9,7 +9,8 @@ module Engine
     module Analyzers
       module Python
         class Main
-          DEFAULT_MASS_THRESHOLD = 50
+          DEFAULT_MASS_THRESHOLD = 40
+          BASE_POINTS = 1000
```
Comment (GordonDiggs) [inline lib/cc/engine/analyzers/python/main.rb]: Why are the base points different for python?

Comment (BlakeWilliams) [inline lib/cc/engine/analyzers/python/main.rb]: The s-expressions generated from Python's AST is much more verbose than Ruby's which means the points have to be lower.

Comment (GordonDiggs) [inline lib/cc/engine/analyzers/python/main.rb]: Cool - makes sense to me!

**Comment by BlakeWilliams** [comment]
Re-review please? cc @mxie

File: lib/cc/engine/analyzers/python/node.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ module Engine
     module Analyzers
       module Python
         class Node < CC::Engine::Analyzers::Node
-          SCRUB_PROPERTIES = ["_type", "attributes"].freeze
+          SCRUB_PROPERTIES = ["_type", "attributes", "ctx"].freeze
```
Comment (GordonDiggs) [inline lib/cc/engine/analyzers/python/node.rb]: :lipstick: Can you make this a `%w[]` array?

Comment (BlakeWilliams) [inline lib/cc/engine/analyzers/python/node.rb]: Yep! That'll look much nicer.

**Comment by GordonDiggs** [comment]
One minor nit but otherwise LGTM

**Comment by BlakeWilliams** [comment]
Thanks!
